### DISCUSSION TITLE
Remix local dev server very slow and takes too much memory and CPU

### DISCUSSION THREAD

**Discussion by kishanhitk** [remix-run/remix#5234] [Q&A]
I am currently facing issue with my remix app setup with cloudflare-pages, when working on the local development server. The esbuild process is consuming a significant amount of memory (30 GBs+) and CPU resources (~477+ %), which is causing my M1 MacBook Air to lag significantly. The live reload feature is also slow and unresponsive, taking around 1 minute for the browser to refresh and show the latest changes made to the code. This issue is not limited to just my device, as my teammates who are using Linux and Windows are also facing the same problem. My current workflow for addressing this issue is to shut down the development server, make code changes, and then restart the server, which is highly inefficient and time-consuming. 

I am looking for potential reasons behind this issue and seeking solutions to improve the overall performance of the development process.

Also getting this in the logs
```
▲ [WARNING] Top-level "this" will be replaced with undefined since this file is an ECMAScript module [this-is-undefined-in-esm]

    functions/[[path]].js:34212:2:
      34212 │   this
            │   ~~~~
            ╵   undefined

  This file is considered to be an ECMAScript module because of the "export" keyword here:

    functions/[[path]].js:210429:0:
      210429 │ export {
             ╵ ~~~~~~
             
```

![Screenshot 2023-01-18 at 8 37 27 PM](https://user-images.githubusercontent.com/41117038/214324678-2cb7de2d-e597-4a13-9cd7-a24c295ad136.png)
![Screenshot 2023-01-24 at 6 28 06 PM](https://user-images.githubusercontent.com/41117038/214324646-4cfb2ac4-9245-4409-bff8-6abd0559c11c.png)
My package.json details:
```
    "dependencies": {
        "@remix-run/cloudflare": "^1.11.1",
        "@remix-run/cloudflare-pages": "^1.11.1",
        "@remix-run/react": "^1.11.1",
        "@supabase/auth-helpers-remix": "^0.1.4",
        "@supabase/supabase-js": "^2.4.1",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
        "tailwindcss": "^3.2.4",
        "wrangler": "^2.8.1",


        .........
    },


        "scripts": {
        "dev:remix": "remix watch",
        "dev:wrangler": "node ./env.js",
        "dev:css": "tailwindcss -w -i ./styles/app.css -o app/styles/app.css",
        "dev": "remix build && run-p \"dev:*\"",
        ....
    },
```

My `remix.config.js`
```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
    serverBuildTarget: "cloudflare-pages",
    server: "./server.js",
};
```

**Comment by machour** [comment]
Hey @kishanhitk 

This might be related to your problem, and a fix for it have been merged yesterday: https://github.com/remix-run/remix/issues/3123
The fix should be available in the next Remix release, which I think will be released today or tomorrow. Stay tuned!
